src/conn.rs

```diff
@@ -705,6 +729,8 @@ impl Conn {
             schema: (*current_schema).clone(),
             cache: InProgressSQLiteAttributeCache::from_cache(cache_cow),
             use_caching: true,
+            tx_reports: Vec::new(),
```

Comment:
I'm tempted to use `smallvec` for this. I imagine most `InProgress`es will do 0-3 `transact`s.